packages/providers/src/script.ts

```diff
@@ -103,17 +103,16 @@ export class Script<TData = void, TResult = void> {
   }
 
   getScriptDataOffset() {
-    return (
-      VM_TX_MEMORY + TRANSACTION_SCRIPT_FIXED_SIZE + ASSET_ID_LEN + AMOUNT_LEN + this.bytes.length
-    );
+    return VM_TX_MEMORY + TRANSACTION_SCRIPT_FIXED_SIZE + this.bytes.length;
```

Comment:
`ASSET_ID_LEN + AMOUNT_LEN` is always required on the contract call offset I think, we should keep both here.

As `getArgOffset()` its ysed for struct inputs and `ASSET_ID_LEN + AMOUNT_LEN` are required on all calls.

Reply:
Makes sense.

I think some thoughts on this;
- Maybe we should create another method that gives the `offset` where the data should be placed.
- With `lw` we are able to pass values bigger than `u64` to `OP:CALL` and this offsets the data to go after `amount`
   - When forwarding `asset_id` we also add this to the offset, even when the params of the `call` are not of the type struct